web3/eth.py

```diff
@@ -469,7 +470,14 @@ def blockNumber(self) -> BlockNumber:
 
     @property
     def chain_id(self) -> int:
-        return self._chain_id()
+        if self._default_chain_id is None:
+            return self._chain_id()
```

Comment:
Would it make sense to set `_default_chain_id` here before returning the value?

Reply:
Yes, it totally would! (That's what I meant with: "As a further improvement, it may be beneficial to cache the result of the first eth_chainId request.")

But that would also change the current behavior and I don't know if that's something that is wanted.